pandas/core/groupby/ops.py

```diff
@@ -475,6 +477,12 @@ def _cython_operation(self, kind, values, how, axis, min_count=-1, **kwargs):
                     "timedelta64 type does not support {} operations".format(how)
                 )
 
+        if is_datetime64tz_dtype(values.dtype):
+            # Cast to naive; we'll cast back at the end of the function
+            # TODO: possible need to reshape?  kludge can be avoided when
+            #  2D EA is allowed.
+            values = values.view("M8[ns]")
+
```

Comment:
can you move the is_datetimelike and is_numeric checks up here and de-duplicate as needed

Reply:
done.  not much de-duplication available.  in another pass we can separate out some of the casting from the everything-else and see if there is de-duplication available in algos or something

Reply:
ok fair, that routine likely needs some cleaning at some point